### PR TITLE
Secure guardrails conceptual documentation

### PR DESCRIPTION
**Pull request by s-santillan**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [ ] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs
- [ ] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: docs/contributing/contributing-rules.md [hunk 1]
```diff
@@ -458,16 +460,15 @@ subcategory:
 
 ##### secure default
 
-A secure default rule is useful for companies writing custom rules. For example, finding all usages to non-standard XML parsing libraries within the company. The rule can also bring a message that a developer can use only a company-approved library.
-
+A secure default rule makes use of inherently secure libraries, frameworks, configurations, or settings. These rules enforce the mitigation of common security concerns, such as preventing cross-site request forgery (CSRF) by properly verifying inbound requests in Django or Flask applications.
```
Comment (s-santillan) [inline docs/contributing/contributing-rules.md:463]: The only actual change that is **not** a vale compliance run.

File: docs/contributing/contributing-rules.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ import RequiredRuleFields from "/src/components/reference/_required-rule-fields.
 
 # Contributing rules
 
-Publish rules in the open-source Semgrep Registry and share them with the Semgrep community to help others benefit from your rule-writing efforts and contribute to the field of software security. There are two ways in which you can contribute rules to the Semgrep Registry:
```
Comment (s-santillan) [inline docs/contributing/contributing-rules.md:14]: Most of these changes are **vale compliance** fixes.

File: docs/secure-guardrails/custom-guardrails-rules.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+---
```
Comment (s-santillan) [inline docs/secure-guardrails/custom-guardrails-rules.md:1]: This doc was copied directly from the `s-santillan` fork except for 1 change (I will mark).

File: docs/secure-guardrails/overview.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
+---
```
Comment (s-santillan) [inline docs/secure-guardrails/overview.md:1]: This doc was copied directly from the `s-santillan` fork except for 1 change (I will mark).

File: docs/secure-guardrails/overview.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+### Remediation guidance
+
+Remediation guidance can come in three forms:
+
+- The rule's `message`
+- AI-generated remediation guidance through Semgrep Assistant
+- The rule's `fix`
+
+Much of the remediation guidance originates from the rule itself, which is also used to generate Semgrep Assistant's advice (if Assistant is enabled). Learning the basic Semgrep rule structure can help you:
```
Comment (s-santillan) [inline docs/secure-guardrails/overview.md:139]: @pabloest Addresses your original note in previous PR:

> My thought was that going down the path of learning the rule structure is to go down the path of making entirely custom rules, free from AI. I see the point that learning the structure is valuable for AI-assisted things, too.

**This is the only change from the last PR's doc.**

**Review by khorne3 — APPROVED** [review]

**Review by pabloest — APPROVED** [review]

**Pull request merged**